File: 0238-product-of-array-except-self/0238-product-of-array-except-self.ts
function productExceptSelf(nums: number[]): number[] {
    let ans = Array(nums.length).fill(1)
    
    let left: number = 1
    for (let i=0; i<nums.length; i++) {
        ans[i] = left
        left *= nums[i]
    }

    let right: number = 1
    for (let i=nums.length -1; i >= 0; i--) {
        ans[i] *= right
        right *= nums[i]
    }
    
    return ans
};

File: 0238-product-of-array-except-self/0238-product-of-array-except-self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ans = [1 for i in range(len(nums))]

        left = 1
        for i in range(len(nums)):
            ans[i] *= left
            left *= nums[i]

        right = 1
        for i in reversed(range(len(nums))):
            ans[i] *= right
            right *= nums[i]

        return ans